design-toolkit/components/src/components/date-field/index.tsx

```diff
@@ -155,51 +108,27 @@ const DateInput = ({
     <div className='relative flex'>
       {size === 'medium' ? (
         <Icon
-          className={cn([
-            '-translate-y-1/2 absolute top-1/2 left-s',
-            props.isDisabled ? 'text-disabled' : 'text-default-light',
-          ])}
+          className={`-translate-y-1/2 absolute top-1/2 left-s ${props.isDisabled ? ' text-disabled' : ' text-default-light'}`}
         >
           <Calendar />
         </Icon>
       ) : null}
       <AriaDateInput
         {...props}
-        className={({ isDisabled, isInvalid }) =>
-          cn(
-            dateFieldStyles({
-              isDisabled,
-              isInvalid,
-              isReadOnly: isReadOnly,
-              size,
-              className,
-            }),
-          )
-        }
+        className={composeRenderProps(className, (className) =>
+          DateFieldStyles({
+            isDisabled: props.isDisabled,
```

Comment:
Instead of these being variants passed into `tv`, you can use the custom variants `disabled:` `invalid:` etc,

Source:
https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/variants/variants.css

Usage:
https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/components/button/styles.ts